modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oidc/OIDCAuthzCodeIdTokenValidationWhenSessionExistsTestCase.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ *
+ */
+
+package org.wso2.identity.integration.test.oidc;
+
+import org.apache.http.HttpResponse;
+import org.apache.http.NameValuePair;
+import org.apache.http.message.BasicNameValuePair;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+import org.wso2.identity.integration.test.utils.DataExtractUtil;
+import org.wso2.identity.integration.test.utils.OAuth2Constant;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class OIDCAuthzCodeIdTokenValidationWhenSessionExistsTestCase extends OIDCAuthzCodeIdTokenValidationTestCase {
+
+
+    @Test(groups = "wso2.is", description = "Send authorize user request for authorization code grant type.")
+    public void testNewAuthCodeGrantSendAuthRequestPost() throws Exception {
+
+        testSecondAuthCodeGrantSendAuthRequestPost();
+        testAuthCodeGrantSendApprovalPost();
+        testAuthCodeGrantSendGetTokensPost();
```

Comment:
shouldn't these methods be individual tests?

Reply:
They themselves are test-cases mentioned in supper class,  So is it intercessory to make it individual test cases again?